src/mono/mono/mini/mini-amd64.c

```diff
@@ -1971,6 +2000,11 @@ mono_arch_allocate_vars (MonoCompile *cfg)
 
 				break;
 			}
+			case ArgSwiftError: {
+					ins->flags &= ~MONO_INST_IS_DEAD;
```

Comment:
Why is this needed ?

Reply:
The SwiftError is marked as a dead arg and could be skipped in the method prolog:
```c
if (ins->flags & MONO_INST_IS_DEAD && !MONO_CFG_PROFILE (cfg, ENTER_CONTEXT))
	/* Unused arguments */
	continue;
```
The `cfg->arch.swift_error_var->flags |= MONO_INST_IS_DEAD;` is used in the method epilog. The runtime should store R12 into the SwiftError only if the arg is passed. On x64, the `swift_error_var` is always allocated, so this iteration sets MONO_INST_IS_DEAD flag for swift_error_var which is then checked in the method epilog.

Reply:
The variable is marked as VOLATILE when its created, so it shouldn't end up dead.

Reply:
The `cfg->arch.swift_error_var` is marked as volatile, but (arg) ins is not.

The `cfg->arch.swift_error_var` utilizes the MONO_INST_IS_DEAD to indicate that the arg is present in the signature. Is it a good idea to introduce a new flag for that instead?

Reply:
The ArgSwiftError ins is marked as dead before the `mono_arch_allocate_vars`.

Reply:
I would avoid iterating over args in `mono_arch_create_vars`. Do you think it is possible?

Reply:
You can store the argument index of the swifterror argument in CallInfo, and the use that in create_vars () to index into cfg->args.